libraries/package-deps-hash/src/getRepoState.ts

```diff
@@ -290,10 +302,53 @@ export function getRepoChanges(
   );
 
   if (result.status !== 0) {
+    const gitVersion: IGitVersion = getGitVersion(gitPath);
+    if (
+      gitVersion.major < MINIMUM_GIT_VERSION.major ||
+      (gitVersion.major === MINIMUM_GIT_VERSION.major && gitVersion.minor < MINIMUM_GIT_VERSION.minor) ||
+      (gitVersion.major === MINIMUM_GIT_VERSION.major &&
+        gitVersion.minor === MINIMUM_GIT_VERSION.minor &&
+        gitVersion.patch < MINIMUM_GIT_VERSION.patch)
+    ) {
+      throw new Error(
+        `The minimum Git version required is ` +
+          `${MINIMUM_GIT_VERSION.major}.${MINIMUM_GIT_VERSION.minor}.${MINIMUM_GIT_VERSION.patch}. ` +
+          `Your version is ${gitVersion.major}.${gitVersion.minor}.${gitVersion.patch}.`
+      );
+    }
+
     throw new Error(`git diff-index exited with status ${result.status}: ${result.stderr}`);
   }
 
   const changes: Map<string, IFileDiffStatus> = parseGitDiffIndex(result.stdout);
 
   return changes;
 }
+
+export function getGitVersion(gitPath?: string): IGitVersion {
+  const result: child_process.SpawnSyncReturns<string> = Executable.spawnSync(gitPath || 'git', ['version']);
+
+  if (result.status !== 0) {
+    throw new Error(`git version exited with status ${result.status}: ${result.stderr}`);
+  }
+
+  return parseGitVersion(result.stdout);
+}
+
+export function parseGitVersion(gitVersionOutput: string): IGitVersion {
+  const versionRegex: RegExp = /^git version (\d+)\.(\d+)\.(\d+)/;
+  const match: RegExpMatchArray | null = versionRegex.exec(gitVersionOutput);
+  if (!match) {
+    throw new Error(`Unable to parse git version output: ${gitVersionOutput}`);
```

Comment:
According to Wikipedia, not all Git version numbers conform to the A.B.C syntax:

https://en.wikipedia.org/wiki/Git#Releases

Reply:
Is the only one that won't be successfully parsed the version released in 2005?

Reply:
But they do seem to all conform to the pattern of `<MAJOR>.<MINOR>.<arbitrary additional characters>`.

Reply:
Perhaps the version check only needs to consider MAJOR/MINOR parts?

Reply:
👍